libcontainer/cgroups/systemd/v2.go

```diff
@@ -174,7 +174,11 @@ func unifiedResToSystemdProps(cm *dbusConnManager, res map[string]string) (props
 	return props, nil
 }
 
-func genV2ResourcesProperties(r *configs.Resources, cm *dbusConnManager) ([]systemdDbus.Property, error) {
+func genV2ResourcesProperties(dirPath string, r *configs.Resources, cm *dbusConnManager) ([]systemdDbus.Property, error) {
+	if err := fs2.CheckMemoryUsage(dirPath, r); err != nil {
```

Comment:
We need this check before setting systemd properties, otherwise the container is OOM-killed and the systemd unit is removed before we get to `fsMgr.Set()`